language-support/java/bindings/src/main/java/com/daml/ledger/javaapi/data/CommandsSubmission.java

```diff
@@ -0,0 +1,246 @@
+// Copyright (c) 2022 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.ledger.javaapi.data;
+
+import static java.util.Collections.emptyList;
+import static java.util.Collections.unmodifiableList;
+import static java.util.Optional.empty;
+
+import com.daml.ledger.javaapi.data.codegen.HasCommands;
+import java.time.Duration;
+import java.time.Instant;
+import java.util.List;
+import java.util.Optional;
+import org.checkerframework.checker.nullness.qual.NonNull;
+
+/**
+ * This class should be used to build a valid submission. It provides {@link #create(String, String, List)}
+ * for initial creation and extra builder methods for additional parameters
+ * e.g {@link #withActAs(List)}, {@link #withWorkflowId(String)} etc.
+ *
+ * Usage:
+ * <pre>
+ *   var params = CommandsSubmission.create("applicationId", "commandId", commands)
+ *                                   .withAccessToken("token")
+ *                                   .withParty("party")
+ *                                   .with...
+ * <pre/>
+ */
+public final class CommandsSubmission {
+  private String applicationId;
+  private String commandId;
+  private List<@NonNull ? extends HasCommands> commands;
+
+  private Optional<String> workflowId;
+  private List<@NonNull String> actAs;
+  private List<@NonNull String> readAs;
+  private Optional<Instant> minLedgerTimeAbs;
+  private Optional<Duration> minLedgerTimeRel;
+  private Optional<Duration> deduplicationTime;
+  private Optional<String> accessToken;
+
+  private CommandsSubmission(
```

Comment:
Minor, order the fields here in constructor and definition from mandatory for create, then lists, then optionals. (I'm in a nitpicky mood today, bear with me!)